src/platform_impl/windows/event_loop.rs

```diff
@@ -124,12 +125,39 @@ static GET_POINTER_TOUCH_INFO: Lazy<Option<GetPointerTouchInfo>> =
 static GET_POINTER_PEN_INFO: Lazy<Option<GetPointerPenInfo>> =
     Lazy::new(|| get_function!("user32.dll", GetPointerPenInfo));
 
+struct Userdata<T: 'static> {
+    recurse_depth: Cell<u32>,
+    removed: Cell<bool>,
+    window_data: Option<WindowData<T>>,
+}
+
+impl<T> Userdata<T> {
+    fn new() -> Self {
+        Self {
+            recurse_depth: Cell::new(0),
+            removed: Cell::new(false),
+            window_data: None,
+        }
+    }
+
+    fn increment_recurse_depth(&self) {
```

Comment:
make it return the depth like `decrement_recurse_depth` does ?

Reply:
It would perhaps be nice for symmetry, but we only really need to check the recurse depth after calling `decrement_recurse_depth`.